nmostesting/mocks/Node.py

```diff
@@ -566,33 +567,52 @@ def transport_file(version, resource, resource_id):
                 interlace = "interlace; "
+
             # TODO: The SDP_PREFERENCES doesn't include video media type
             sdp_file = template.render(dst_ip=destination_ip,
                                        dst_port=destination_port,
                                        src_ip=source_ip,
-                                       media_type="raw",
-                                       width=CONFIG.SDP_PREFERENCES["video_width"],
-                                       height=CONFIG.SDP_PREFERENCES["video_height"],
+                                       media_type=sdp_params.get("media_type", "raw"),
+                                       width=sdp_params.get("video_width",
+                                                            CONFIG.SDP_PREFERENCES["video_width"]),
+                                       height=sdp_params.get("video_height",
+                                                             CONFIG.SDP_PREFERENCES["video_height"]),
                                        interlace=interlace,
-                                       exactframerate=CONFIG.SDP_PREFERENCES["video_exactframerate"],
-                                       depth=CONFIG.SDP_PREFERENCES["video_depth"],
-                                       sampling=CONFIG.SDP_PREFERENCES["video_sampling"],
-                                       colorimetry=CONFIG.SDP_PREFERENCES["video_colorimetry"],
-                                       transfer_characteristic=CONFIG.SDP_PREFERENCES["video_transfer_characteristic"],
-                                       type_parameter=CONFIG.SDP_PREFERENCES["video_type_parameter"])
+                                       exactframerate=sdp_params.get("video_exactframerate",
+                                                                     CONFIG.SDP_PREFERENCES["video_exactframerate"]),
+                                       depth=sdp_params.get("video_depth",
+                                                            CONFIG.SDP_PREFERENCES["video_depth"]),
+                                       sampling=sdp_params.get("video_sampling",
+                                                               CONFIG.SDP_PREFERENCES["video_sampling"]),
+                                       colorimetry=sdp_params.get("video_colorimetry",
+                                                                  CONFIG.SDP_PREFERENCES["video_colorimetry"]),
+                                       transfer_characteristic=sdp_params.get("video_transfer_characteristic",
+                                                                              CONFIG.SDP_PREFERENCES[
+                                                                                   "video_transfer_characteristic"]),
+                                       type_parameter=sdp_params.get("video_type_parameter",
+                                                                     CONFIG.SDP_PREFERENCES["video_type_parameter"]),
```

Comment:
This looks like it would be simpler if you constructed `sdp_params` by starting with `CONFIG.SDP_PREFERENCES` and then inserting `sender.get('sdp_params', {})` so that configured preferences got overridden with specific values from the sender?

Something like:

```python
sdp_params = {**CONFIG.SDP_PREFERENCES, **sender.get('sdp_params', {})}
```

FWIW, from Python 3.9 I think we could use the merge operator:

```python
sdp_params = CONFIG.SDP_PREFERENCES | sender.get('sdp_params', {})
```

Reply:
That's a  much cleaner idea. There's a lot of the same type of template manipulation in the mock Node - it feels like it should be factored into helper functions...